app/utils/client-hints.tsx

```diff
@@ -0,0 +1,107 @@
+import { type Request } from '@remix-run/node'
+
+const colorSchemeHint = {
+	name: 'CH-prefers-color-scheme',
+	getValueCode: `window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light'`,
+	fallback: 'light',
+	transform(value: any) {
+		return value === 'dark' ? 'dark' : 'light'
+	},
+} as const
+
+const clockOffsetHint = {
```

Comment:
I'm not sure clock offset alone is sufficient for rendering dates on the server. One may need to render a localized date/time, including the timezone. e.g. `8:00 AM PDT` - for this, we would need to know the user's IANA timezone (which would also tell us the appropriate clock offset).

The user's timezone is a little more resilient in general to handle transitions between daylight saving time as well.

Reply:
Thanks! That's fair. I'm still deciding whether I need to include this in the template. I'm pretty sure all the template needs to have built-in is the color preference for dark mode. The timezone stuff can probably just be docs.

Reply:
Ya, the docs are helpful either way to be clear how folks can extend to add their own app-specific preferences. 👍

Reply:
I would certainly be grateful for a PR for docs like this once this is merged :)